include/cudaq/Optimizer/Dialect/Quake/QuakeOps.td

```diff
@@ -1397,4 +1397,89 @@ def CustomUnitarySymbolOp :
   }];
 }
 
+//===----------------------------------------------------------------------===//
+// Quantum states
+//===----------------------------------------------------------------------===//
+
+def quake_CreateStateOp : QuakeOp<"create_state", [Pure] > {
+  let summary = "Create state from data";
+  let description = [{
+    This operation takes a pointer to state data and creates a quantum state.
+    The operation can be optimized away in DeleteStates pass, or replaced
+    by an intrinsic runtime call on simulators.
+
+    ```mlir
+      %0 = quake.create_state %data %len: !cc.ptr<!cc.state>
```

Comment:
This example doesn't correspond to the assembly syntax below.

Reply:
fixed, thanks!